src/docbuilder/rustwide_builder.rs

```diff
@@ -535,50 +516,181 @@ impl RustwideBuilder {
+                                    total += n.as_i64().unwrap_or(0) as i32;
+                                }
+                                if let Some(Value::Number(n)) = entry.get("with_docs") {
+                                    documented += n.as_i64().unwrap_or(0) as i32;
+                                }
+                            }
+                            Ok(Some((total, documented)))
+                        }
+                        _ => Ok(None),
+                    }
+                } else {
+                    Ok(None)
+                }
+            },
+        )
+    }
+
+    fn execute_build(
+        &self,
+        target: &str,
+        is_default_target: bool,
+        build: &Build,
+        limits: &Limits,
+        metadata: &Metadata,
+    ) -> Result<FullBuildResult> {
+        self.make_build_object(
+            target,
+            metadata,
+            limits,
+            |cargo_args, mut rustdoc_flags, storage| {
+                rustdoc_flags.extend(vec![
+                    "--resource-suffix".to_string(),
+                    format!("-{}", parse_rustc_version(&self.rustc_version)?),
+                ]);
+                let cargo_metadata = CargoMetadata::load(
+                    &self.workspace,
+                    &self.toolchain,
+                    &build.host_source_dir(),
+                )?;
+
+                for dep in &cargo_metadata.root_dependencies() {
+                    rustdoc_flags.push("--extern-html-root-url".to_string());
+                    rustdoc_flags.push(format!(
+                        "{}=https://docs.rs/{}/{}",
+                        dep.name.replace("-", "_"),
+                        dep.name,
+                        dep.version
+                    ));
+                }
+                if let Some(package_rustdoc_args) = &metadata.rustdoc_args {
+                    rustdoc_flags
+                        .append(&mut package_rustdoc_args.iter().map(|s| s.to_owned()).collect());
+                }
+
+                let successful = logging::capture(&storage, || {
+                    build
+                        .cargo()
+                        .timeout(Some(limits.timeout()))
+                        .no_output_timeout(None)
+                        .env(
+                            "RUSTFLAGS",
+                            metadata
+                                .rustc_args
+                                .as_ref()
+                                .map(|args| args.join(" "))
+                                .unwrap_or_default(),
+                        )
+                        .env("RUSTDOCFLAGS", rustdoc_flags.join(" "))
+                        // For docs.rs detection from build script:
+                        // https://github.com/rust-lang/docs.rs/issues/147
+                        .env("DOCS_RS", "1")
+                        .args(&cargo_args)
+                        .run()
+                        .is_ok()
```

Comment:
This same sequence comes up a lot, could you pull it into a method on `DocBuilder`?

Reply:
The sequence is quite small and can only be found in two places. Is it really worth it considering that even though they look similar, they have differences?

Reply:
I guess it doesn't matter a ton, but some of the common stuff could be (like setting the env vars) so that there's only one place to change

Reply:
I would rather have this in one place so that people don't accidentally update it in only one place. This is the part of docs.rs that has the fewest tests so it should be as simple to modify as possible.

Reply:
Fair enough. I'll try to move code into a function.